pkg/provider/conjur/provider.go

```diff
@@ -0,0 +1,159 @@
+// Package conjur provides a Conjur provider for External Secrets.
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package conjur
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/cyberark/conjur-api-go/conjurapi"
+	"github.com/cyberark/conjur-api-go/conjurapi/authn"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	esv1beta1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
+)
+
+var (
+	errConjurClient          = "cannot setup new Conjur client: %w"
+	errMissingStore          = fmt.Errorf("missing store provider")
+	errMissingConjurProvider = fmt.Errorf("missing store provider Conjur")
+)
+
+// Provider is a provider for Conjur.
+type Provider struct {
+	ConjurClient Client
+}
+
+// Client is an interface for the Conjur client.
+type Client interface {
+	RetrieveSecret(secret string) (result []byte, err error)
+}
+
+// NewClient creates a new Conjur client.
+func (p *Provider) NewClient(ctx context.Context, store esv1beta1.GenericStore, kube client.Client, namespace string) (esv1beta1.SecretsClient, error) {
+	cfg, err := getProvider(store)
+	if err != nil {
+		return nil, err
+	}
+	config := conjurapi.Config{
+		Account:      *cfg.ServiceAccount,
+		ApplianceURL: *cfg.ServiceURL,
+		SSLCert:      *cfg.ServiceCertificate,
```

Comment:
The `cfg.ServiceCertificate` can be `nil` and can throw a `nil pointer dereference` error.
It's probably best to make it a `string` instead of `*string`, as it seems the SDK handles that case well (empty string => no TLS).